samples/caf/src/modules/led_state.c

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr.h>
+
+#define MODULE led_state
+#include <caf/events/module_state_event.h>
+#include <caf/events/led_event.h>
+#include <caf/events/button_event.h>
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(MODULE, CONFIG_LOG_DEFAULT_LEVEL);
+
+#include "led_state_def.h"
+
+
+enum button_id {
+	BUTTON_ID_NEXT_EFFECT,
+	BUTTON_ID_NEXT_LED,
+
+	BUTTON_ID_COUNT
+};
+
+static enum led_effect_id curr_led_effect_id[LED_ID_COUNT];
+static enum led_id curr_led_id;
+
+
+static void send_led_event(enum led_id led_id, const struct led_effect *led_effect)
+{
+	__ASSERT_NO_MSG(led_effect);
+	__ASSERT_NO_MSG(led_id < LED_ID_COUNT);
+
+	struct led_event *event = new_led_event();
+
+	event->led_id = led_id;
+	event->led_effect = led_effect;
+	EVENT_SUBMIT(event);
+}
+
+static bool handle_button_event(const struct button_event *evt)
+{
+	ARG_UNUSED(evt);
+
+	if (evt->pressed) {
+		switch (evt->key_id) {
+		case BUTTON_ID_NEXT_LED:
+			/* Send led effect for current LED. */
+			send_led_event(curr_led_id, &led_effect[curr_led_effect_id[curr_led_id]]);
+
+			/* Control next LED, signalize by turning on. */
+			if (++curr_led_id == LED_ID_COUNT) {
+				curr_led_id = LED_ID_0;
+			}
+			send_led_event(curr_led_id, &led_effect_on);
+			break;
+		case BUTTON_ID_NEXT_EFFECT:
+			/* Change and display new effect. */
+			if (++curr_led_effect_id[curr_led_id] == LED_EFFECT_ID_COUNT) {
+				curr_led_effect_id[curr_led_id] = LED_EFFECT_ID_OFF;
```

Comment:
Consider adding BUILD_ASSERT to ensure that LED_EFFECT_ID_OFF == 0 (so that we will actually cycle through all of the LED effects also when user redefines them as suggested in testing section)

Reply:
Good point.

Reply:
You could place the BUILD_ASSERT right above this lane, so it will be obvious why it was added